include/engine/datafacade/algorithm_datafacade.hpp

```diff
@@ -0,0 +1,67 @@
+#ifndef OSRM_ENGINE_DATAFACADE_ALGORITHM_DATAFACADE_HPP
+#define OSRM_ENGINE_DATAFACADE_ALGORITHM_DATAFACADE_HPP
+
+#include "contractor/query_edge.hpp"
+#include "engine/algorithm.hpp"
+#include "util/integer_range.hpp"
+
+namespace osrm
+{
+namespace engine
+{
+namespace datafacade
+{
+
+using EdgeRange = util::range<EdgeID>;
```

Comment:
Seeing this, I'd really love our graphs moving away from the manual node and edge iteration and provide a proper iterator / range based interface. Would be great to be able to say

    for (auto node in graph.Nodes())
        for (auto edge in graph.Edges(node))
            fn(edge);

The partition graph already has something similar to this:

https://github.com/Project-OSRM/osrm-backend/blob/6f34d70907dadb225d8a2e8bc70827275e991fc6/include/partition/partition_graph.hpp

(It's out of context here but it just came to mind while looking at the EdgeRange)